File: js/src/0.0-Basics/0.3-build-up-logical-thinking/0.3.18-pattern.js
function log(val) {
    process.stdout.write(String(val));
}

function pattern_18() {
    let chars = ['a', 'b', 'c', 'd', 'e'];

    let row = 5;
    let column = 10;

    for (let y = 0; y < row; y++) {
        for (let x = 1; x < column + 1; x++) {
            if (x <= column / 2) {
                if (x > y) {
                    log(chars[x - 1]);
                } else {
                    log(' ');
                }
            } else if (x - 1 > column / 2) {
                let element = column - x;

                if (element >= y) {
                    log(chars[element]);
                } else {
                    log(' ');
                }
            }
        }

        console.log();
    }
}

console.clear();
pattern_18();

File: python/00-Basics/00.03-build-up-logical-thinking/0.3.18-pattern.py
from utils.clear import clear

def pattern_18():
    row = 5
    column = 10

    chars = ['a', 'b', 'c', 'd', 'e']

    for i in range(0, row ):
        for j in range(0, column + 1):
            if j < column / 2:
                if j >= i:
                    print(chars[j], end="")
                elif j < i:
                    print(" ", end="")
            elif j > column / 2:
                key = column - j - 1
                if key >= i:
                    print(chars[key], end="")
                elif key < i:
                    print(" ", end="")

        print()



clear()
pattern_18()
